src/main/groovy/com/calgaryscientific/gradle/VeracodeTask.groovy

```diff
@@ -44,50 +44,45 @@ abstract class VeracodeTask extends DefaultTask {
             'fileId'           : '123',
             'mode'             : 'action|actionSummary|verbose',
             'maxUploadAttempts': '123',
-            'fileId'           : 'xxx',
             'buildId1'         : '123',
             'buildId2'         : '123'
     ]
 
-    def requiredArguments = []
+    List<String> requiredArguments = []
+    List<String> optionalArguments = []
 
     VeracodeTask() {
         group = 'Veracode'
     }
 
     abstract void run()
 
-    final String correctUsage() {
-        StringBuilder sb = new StringBuilder("Example of usage: gradle ${getName()}")
-        requiredArguments.each() {
-            if (!isArgumentOptional(it)) {
-                sb.append(" -P${it}=${validArguments.get(it)}")
-            } else {
-                String originalArgument = it.substring(0, it.length() - OPTIONAL.length())
-                sb.append(" [-P${originalArgument}=${validArguments.get(originalArgument)}]")
-            }
+    protected static String correctUsage(String taskName,
+                                         List<String> requiredArguments,
+                                         List<String> optionalArguments) {
+        StringBuilder sb = new StringBuilder("Missing required arguments: gradle ${taskName}")
+        requiredArguments.each() { arg ->
+            sb.append(" -P${arg}=${validArguments.get(arg)}")
```

Comment:
I would do something with groovy idiom: requiredArguments.collect { arg -> -P${arg}=${validArguments.get(arg)} }.join(' ')

Reply:
It doesn't gain much readability in this case I don't think:

````groovy
        requiredArguments.each() { arg ->
            sb.append(" -P${arg}=${validArguments.get(arg)}")
        }
````
vs
````groovy
        sb.append(" " + requiredArguments.collect { arg ->
            "-P${arg}=${validArguments.get(arg)}"}.join(" ")
        )
````
or
````groovy
        sb.append(" ")
        sb.append(requiredArguments.collect { arg ->
            "-P${arg}=${validArguments.get(arg)}"}.join(" ")
        )
````